bc_obps/registration/api/user_operator.py

```diff
@@ -358,71 +358,38 @@ def create_user_operator_contact(request, payload: UserOperatorContactIn):
 
 
 ##### PUT #####
-
-
-# Function to check if the status is valid. This was required because the previous check only worked for
-# statuses that didn't have two words (e.g. "Pending" and "Approved" worked, but "Changes Requested" didn't)
-def check_status(status: str):
-    for statusEnum in UserOperator.Statuses:
-        if statusEnum == status:
-            return True
-    return False
-
-
-# this endpoint is for updating the status of a user
-@router.put("/select-operator/user-operator/{user_guid}/update-status")
+@router.put("/select-operator/user-operator/update-status", response={200: UserOperatorOut, codes_4xx: Message})
 @authorize(AppRole.get_all_authorized_app_roles(), ["admin"])
-def update_user_operator_user_status(request, user_guid: str):
-    current_admin_user: User = request.current_user
-    # need to convert request.body (a bytes object) to a string, and convert the string to a JSON object
-    payload = json.loads(request.body.decode())
-    status = payload.get("status")
-    if not check_status(status):
-        return 400, {"message": "Invalid status."}
-    user_operator = get_object_or_404(UserOperator, user=user_guid)
-    user_operator.status = status
+def update_user_operator_status(request, payload: UserOperatorStatusUpdate):
+    current_user: User = request.current_user  # irc user or industry user admin
+    if payload.user_guid:  # to update the status of a user_operator by user_guid
+        user_operator = get_object_or_404(UserOperator, user_id=payload.user_guid)
+    elif payload.user_operator_id:  # to update the status of a user_operator by user_operator_id
+        user_operator = get_object_or_404(UserOperator, id=payload.user_operator_id)
+    else:
+        return 404, {"message": "No parameters provided"}
+
+    # We can't update the status of a user_operator if the operator is not approved
+    if user_operator.operator.status != Operator.Statuses.APPROVED:
+        return 400, {"message": "Operator must be approved before approving users."}
```

Comment:
The logic here will need to be modified. In cases where the operator data has been imported from CIIP/SWRS, its status will be "DRAFT" at the time that a user is requesting prime admin. The user will need to have admin access to the operator before they're able to update/confirm the imported data about the Operator, at which point the Operator's status will be set to Pending/Accepted.

(see the "Andrea's brain dumping ground" Miro board for a map of this workflow if you need more context)